pkg/services/pluginsintegration/pluginconfig/request.go

```diff
@@ -18,7 +19,7 @@ import (
 var _ PluginRequestConfigProvider = (*RequestConfigProvider)(nil)
 
 type PluginRequestConfigProvider interface {
-	PluginRequestConfig(ctx context.Context, pluginID string) map[string]string
```

Comment:
Since we use this dependency [here](https://github.com/grafana/grafana-enterprise/blob/main/src/pkg/extensions/datasource/context.go#L95), it's probably not ideal to have anything to do with `pluginstore` over there. 

We could add the `PluginStore` as a dependency here and do a lookup, therefore we keep the interface as is. That means we'll need a new implementation over the enterprise however, which will probably have the exact same behaviour, I think?

Another alternative is maybe to move `PluginRequestConfigProvider` and `PluginInstanceCfg` into `pkg/plugins` as we already have an indication that this is now a useful component not just for Grafana, but for anything needing to communicate with a plugin. Then we can accept a `plugin *plugins.Plugin` here instead.

Reply:
That's a very good point. Let me use a simplified struct here and I'll open a PR there and we can keep the discussion.